src/LzoStream.cs

```diff
@@ -123,166 +126,180 @@ protected virtual int Decode(byte[] buffer, int offset, int count)
             Debug.Assert(count > 0);
             Debug.Assert(DecodedBuffer == null);
             int read;
-            var i = Instruction >> 4;
-            switch (i)
+
+            if (IsFirstInstruction && Instruction > 17)
+            {
+                int length = Instruction - 17;
```

Comment:
In [minilzo](https://github.com/nemequ/lzo/blob/0083878c235a89ef96a009d1ff0b500f3a364e4b/minilzo/minilzo.c#L5446), there's a special case if `length < 4`, which I'm not handling here because I don't know where the equivalent destination is in lzo.net.